include/sharg/parser.hpp

```diff
@@ -903,6 +878,42 @@ class parser
         if (detail::format_parse::is_empty_id(short_id) && detail::format_parse::is_empty_id(long_id))
             throw design_error("Option Identifiers cannot both be empty.");
     }
+
+    //!brief Verify if the configuration given to an sharg::parser::add_option call is valid.
+    template <typename config_type>
+    void verify_option_config(config_type const & config)
+    {
+        if (sub_parser != nullptr)
+            throw design_error{"You may only specify flags for the top-level parser."};
+
+        verify_identifiers(config.short_id, config.long_id);
```

Comment:
Pass config?

Reply:
again I like that I explicitly know that only the ids are needed